examples/grpc_credential_types.cc

```diff
@@ -100,7 +100,7 @@ google::iam::credentials::v1::GenerateAccessTokenResponse UseAccessToken(
     auto token = client.GenerateAccessToken(
         "projects/-/serviceAccounts/" + service_account, /*delegates=*/{},
         /*scope=*/{"https://www.googleapis.com/auth/cloud-platform"}, duration);
-    if (!token) throw std::runtime_error(token.status().message());
+    if (!token) throw token.status();
```

Comment:
Blegh, `clang-tidy` is unhappy about this:

https://storage.googleapis.com/cloud-cpp-community-publiclogs/logs/google-cloud-cpp/10108/7f850251b5795674be31a845ed64d3d5b6d9f4b4/fedora-36-clang-tidy/log-13dfbf57-59cd-4d1a-a9cd-a1f82e42f7ce.txt

This seems to fix the error for me:

```cc
  if (!token) throw std::move(token).status();
```

Make the same change below too.

Reply:
Aside: It looks like there is nothing here to catch these exceptions, so throwing a `Status` will probably result in less information being reported, contrary to the goal.

Reply:
Oh yeah you're right. Should we add something like this towards the end?
https://github.com/googleapis/google-cloud-cpp/blob/69561feebfd679a6546a829764e711bb93d96665/google/cloud/spanner/samples/postgresql_samples.cc#L1068-L1075

Maybe we can skip catching the std::exception as even if it's not caught, we get a
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  Failed to create CurlHandle
Aborted
```
instead of a
```
Failed to create CurlHandle
```

Reply:
Yes, something like that.

I'm not sure of the sensibilities of skipping the `std::exception` handler.  You're correct that we see the same information either way, but I reckon the answer depends on whether we think of "Failed to create CurlHandle" (or any of the other `runtime_error` cases) as an `exit(1)` or an `abort()`.

I'd vote for a normal termination, albeit indicating an error (i.e., `exit(1)`), but lets see if anyone else has an opinion.

Reply:
What @devbww said sounds good to me.